Assets/MRTK/SDK/Features/Utilities/Solvers/HandConstraint.cs

```diff
@@ -545,6 +589,38 @@ private static Ray CalculateProjectedSafeZoneRay(Vector3 origin, Transform targe
             return new Ray(origin + direction, -direction);
         }
 
+        private static Ray CalculateProjectedSafeZoneRay(
```

Comment:
Design question; is the overloading pattern ideal here? It seems as though the additional overload does significantly different work here. The angle offset _could_ be applied to the AtopPalm safezone, just as a simple rotation around the palm joint "up" axis. If we used the angle offset to rotate the AtopPalm safezone, we could unify these calls and simplify this "special case" (as there would be no more special case; AtopPalm would simply be rotated, too.)

The use case for rotating the AtopPalm constraint could be quite important, actually; when I was designing the Trackbox prototype, I really wanted to have a handedness-dependent rotation offset. The existing HandBounds lets you specify a universal rotation offset already (to the hand constraint itself) but it does not let you do it a handedness-dependent way (i.e. flip the rotation depending on left/right hand). I for one would definitely use this feature if it applied to AtopPalm, and I would guess many others might too.

Reply:
Added support for rotation when safezone is atop palm, refactored the code to not use overloaded method, and added test cases, thanks for the feedback!